Feat: Added new default user; Closes: #637

9921622 opened the pull request:
issue #638 might possibly have been fixed here if you were talking about default teacher for initial blocks and not create blocks.

changed .env file. Not sure why git doest detect it but all i did was change and add:
```
## TENANTS ##############################################

TENANT_DEFAULT_ADMIN_USERNAME=admin
TENANT_DEFAULT_ADMIN_PASSWORD=password

TENANT_DEFAULT_OWNER_USERNAME=owner
TENANT_DEFAULT_OWNER_PASSWORD=password1
```

tylerecouture commented:
@9921622 .env file isn't tracked because it will have all the real secrets/keys in production.  Instead you need to put this in the .env.example (this is what you copied to .env when you set up your development environment)

tylerecouture reviewed (changes requested):
Minor stuff again.

9921622 commented on src/tenant/initialization.py:
```diff
@@ -34,13 +34,21 @@ def load_initial_tenant_data():
+        password=settings.TENANT_DEFAULT_OWNER_PASSWORD,
+    )
+    
+    return owner
```
the returned user will be the default teacher of the default blocks, so only one user model should be returned.

9921622 replied:
its also why I mentioned my commit might have fixed #638 assuming you were talking about the default blocks teacher and not the default teacher for newly created blocks

9921622 replied:
i can probably remove that and instead get the owner using User.models.get(username=settings.TENANT_DEFAULT_OWNER_USERNAME) for the blocks_creation

tylerecouture replied:
Ya that's a good idea, do that.

tylerecouture replied:
However... that's just the initial username, that username will probably be changed by the owner to something more personal, So we can't search by the username or it will only work for the initially created default blocks, not for blocks created later on by the user?

9921622 replied:
im not exactly sure how to add the owner as the default for blocks that are later created. Since if i go by pk then it will only work if the owner doesnt accidentally delete their user model

9921622 commented on src/tenant/tests/test_initialization.py:
```diff
@@ -24,3 +26,23 @@ def test_default_mark_ranges_created(self):
+            
+            # Check if can login
+            success = self.client.login(username=username, password=password)
+            self.assertTrue(success)
```
done

9921622 commented on src/tenant/initialization.py:
```diff
@@ -34,13 +34,21 @@ def load_initial_tenant_data():
 
 
 def create_superuser():
```
done

9921622 commented:
Added requested changes.

Test should work now

9921622 commented:
Test failed because there was a test assuming admin was the default teacher. Fixed

tylerecouture reviewed (changes requested):
Good point about the problem with picking a default user.  Let's try this: get a queryset of staff users, if there are more than 2, get the second one (sorted by id, which is probably default).  If there is only 1, use that.  If there are none, we'll crash errors.  That should go into a function somewhere and have some tests.  Probably best place for this method is the site config app.  

Even better (but a bit more complicated) is to add a field on the site config model for "deck owner" and just use whatever that is set to... actually I'll add that as a new issue, because there will be some special considerations with that.  So for now just create the method `get_deck_owner()` in the siteconfig model and later we can change that method's implementation.

9921622 commented:
This is basically 3 issues in one pr

# issues: #637, #1031 (this pr)
- added new superuser named owner
- owner superuser account info will be in ```.env.example```
- added the requested changes

### only files changed here are
- files that requested change
- ```.env.example```
- ```settings.py```


# issue: #1032
- [x] queryset for this foreignkey field should include active staff users only. help_text should warn indicate that only the current deck owner can change this setting.
- [x] on new decks, set this to the deck owner created during initialization
- [x]  in the site config form, only allow this field to be edited by: deck admin user, or current deck_owner.
- [x] Will probably need a migration to handle existing production decks to populate this field. Easiest way is to set a default on the field, and point it to the "get_deck_owner" method discussed here: https://github.com/bytedeck/bytedeck/pull/1031#pullrequestreview-1007813161

### files
- ```siteconfig\forms.py```
- ```0015_siteconfig_deck_owner.py```
- ```models.py```
- ```views.py```


# issue: #638
- added only one line to ```courses\forms.py``` changing the default teacher for new blocks to deck_owner

### files
- ```courses\forms.py```

tylerecouture reviewed (changes requested):
As I mentioned before, this is too much for a single commit.  Please PR the first issue as it's own commit and own PR and then you can rebase the other changes off of that one.  Because this will effect data on production server, i need it in it's own commit that can be reverted.

9921622 commented:
removed the unrelated changes to #637
